src/s3path.jl

```diff
@@ -355,7 +355,7 @@ function Base.readdir(fp::S3Path; join=false, sort=true)
     end
 end
 
-Base.read(fp::S3Path) = Vector{UInt8}(s3_get(fp.config, fp.bucket, fp.key))
+Base.read(fp::S3Path) = Vector{UInt8}(s3_get(fp.config, fp.bucket, fp.key; raw=true))
```

Comment:
this is the only functional change here, and due to the `Vector{UInt8}` constructor, if one was not getting back raw bytes here, they would get a confusing `MethodError` from the constructor. Now they will get the bytes back.

We saw this often previously when AWS.jl would not throw on redirects and instead send the content of the redirect message back, which would get converted to a LittleDict and error at this point. However, this constructor is not the correct place to check for these things; errors associated to non-OK HTTP status codes should be thrown in `submit_request` with the current design of AWS.jl.